xlsx_reader.py

```diff
@@ -0,0 +1,182 @@
+import xlrd
+
+from nameparser import HumanName
+
+from models import (Book, Author, Copy, Magazine)
+from init_db import db
+
+# please ensure if you have a proper file in your folder
+data = './biblioteka_probna.xlsx'
+workbook = xlrd.open_workbook(data)
+
+
+def get_first_name(author):
+    name = HumanName(str(author))
+    first_name = (name.first + " " + name.middle).strip()
+    return first_name
+
+
+def get_last_name(author):
+    name = HumanName(str(author))
+    last_name = (name.last + " " + name.suffix).strip()
+    return last_name
+
+
+# checking if element exist, creates new if not
+def get_or_create_library_item(session, model, **kwargs):
+    library_item = session.query(model).filter_by(**kwargs).first()
+    if library_item:
+        return library_item
+    else:
+        instance = model(**kwargs)
+        session.add(instance)
+        session.commit()
+        return instance
+
+
+# reading author's data from file
+def get_authors_data(authors):
+    if (',' in authors and 'Jr.' not in authors) \
+            or (' and ' in authors) \
+            or ('&' in authors):
+        split_authors = \
+            authors.replace(' and ', ',').replace('&', ',').split(',')
+        first_names = []
+        last_names = []
+
+        for auth in split_authors:
+            first_name = get_first_name(auth)
+            first_names.append(first_name)
+            last_name = get_last_name(auth)
+            last_names.append(last_name)
+        authors_names = list(zip(first_names, last_names))
+    else:
+        first_name = get_first_name(authors)
+        last_name = get_last_name(authors)
+        authors_names = (first_name, last_name)
+
+    return authors_names
```

Comment:
I'd make functions for getting authors more compacet

Reply:
thanks for that suggestion and your help in this matter

Reply:
No prob